app/components/Forms/AltDateInput.tsx

```diff
@@ -58,7 +59,7 @@ class AltDateInput extends Component {
     },
   };
 
-  constructor(props) {
+  constructor(props: WidgetProps) {
```

Comment:
Similar to the comment below, a component declared as `const AltTimeInput: React.FunctionComponent<Props> = (props) =>` with the following interface 
```ts
interface Props extends WidgetProps {
  time: boolean;
}
```
would provide better type checking.

I also see another component above, in the same file, which is discouraged.

Instead of using a state, you can use the `useMemo` hook